app/code/community/Oyst/OneClick/controllers/Adminhtml/Oneclick/ActionsController.php

```diff
@@ -77,4 +77,70 @@ public function indexAction()
     {
         $this->_initAction()->renderLayout();
     }
+
+    /**
+     * Download requested log file
+     */
+    public function downloadAction()
+    {
+        $params = Mage::app()->getRequest()->getParams();
+        $fileName = Mage::getBaseDir('var') . DS . 'log' . DS . $params['name'];
+
+        if (file_exists($fileName)) {
+            header('Content-Description: File Transfer');
+            header('Content-Type: application/octet-stream');
+            header('Expires: 0');
+            header('Cache-Control: must-revalidate');
+            header('Pragma: public');
+            header('Content-Disposition: attachment; filename="' . basename($fileName) . '"');
+
+            $file = file($fileName);
```

Comment:
Just to challenge your code, is that the most optimized way to grab some lines of very large log file (e.g. system.log weight  1 Go). I let you argue.